src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -88,14 +88,56 @@ class BasicCrawlerOptions(TypedDict, Generic[TCrawlingContext]):
 
 
 class BasicCrawler(Generic[TCrawlingContext]):
-    """Provides a simple framework for parallel crawling of web pages.
-
-    The URLs to crawl are fed either from a static list of URLs or from a dynamic queue of URLs enabling recursive
-    crawling of websites.
-
-    `BasicCrawler` is a low-level tool that requires the user to implement the page download and data extraction
-    functionality themselves. If we want a crawler that already facilitates this functionality, we should consider using
-    one of its subclasses.
+    """A versatile web crawler for parallel URL fetching with extensive features for web scraping.
+
+    BasicCrawler is a highly customizable and efficient web crawling solution that provides:
+    - Automatic scaling based on available system resources and target website's requirements.
+    - Smart request routing and handling of different types of URLs.
+    - Session management for maintaining persistent connections.
+    - Proxy integration and rotation for avoiding IP-based blocking.
+    - Automatic retries for failed requests with customizable settings.
+    - Built-in statistics tracking for monitoring crawler performance.
+    - Event system for fine-grained control over the crawling process.
+
+    Basic usage:
+    ```python
+    async def handle_request(context: Context) -> None:
+        url = context.request.url
+        response = await context.send_request()
+        # Process the response here
+    crawler = BasicCrawler(
+        request_handler=handle_request,
+    )
+    await crawler.run()
```

Comment:
This is not a recommended way of attaching a handler - typically we use decorators.